openedx/features/course_experience/__init__.py

```diff
@@ -75,7 +75,16 @@
 # Waffle flag to enable relative dates for course content
 RELATIVE_DATES_FLAG = ExperimentWaffleFlag(WAFFLE_FLAG_NAMESPACE, 'relative_dates', __name__, experiment_id=17)
 
-# Waffle flag to enable user calendar syncing
+# .. toggle_name: course_experience.calendar_sync
+# .. toggle_implementation: CourseWaffleFlag
+# .. toggle_default: False
+# .. toggle_description: This course flag enables a course tool (which is a tool that is added on a course home page)
+#   that sends course assignment calendars to course students, whenever they click on the "Subscribe to calendar
+#   updates" button. The email contains an ics attachment that students can then use to sync with their own calendar
+#   apps. In order to enable this, the RELATIVE_DATES_FLAG must be enabled, too.
```

Comment:
1. Consider moving `RELATIVE_DATES_FLAG` sentence to a `toggle_warning` annotation.
2. This makes me feel like we need a doc on what a "course flag" or "CourseWaffleFlag" means. This does not affect this PR directly, but if we had such a doc, maybe we could add a section to the "edx-platform Technical Reference" that provides some overview information about how to use the different toggle types.

Reply:
> Consider moving RELATIVE_DATES_FLAG sentence to a toggle_warning annotation.

I did, but eventually decided not to, because enabling RELATIVE_DATES_FLAG should be part of the normal process for enabling this feature. It was a judgment call, basically.

Reply:
I'd like to discuss this further, because it has been a common case for many toggles/settings.  Here are some possibilities for documenting dependencies:
* `toggle_description`: My fear when reading this is that people may not read the full description if they think they know what a toggle/setting is used for, and would miss this important detail. It is true that they might miss other important details as well. Note that we don't yet have documented support for any formatting, but allowing for bold text (or other formatting) might be another more generic solution.
* `toggle_warning(s)`: For most of the PRs I have been reviewing, I think people naturally were using this more often for depends-on dependencies, and I think this is my current preference for being able to highlight this.
* `toggle_dependencies`: We could introduce a new annotation that specifically calls attention to this kind of detail. I don't love this idea mainly because I think it would lead to even more inconsistencies in the annotations.
* Other?

Once we come to a conclusion, we should update the docs to note the recommendation.  We could also open topics on discuss to allow more people to weigh-in on these topics.

FYI: I don't consider this merge-blocking for this PR, even though I would like to resolve this discussion at some point.

Reply:
Sounds good! I moved the information to `.. toggle_warnings:`.